README.md

```diff
@@ -130,7 +130,13 @@ namespace {
 for (const auto plugin : linktimeplugin::plugins<PluginBase>()) {
     plugin->dosomething();
 }
-
+// OR
+std::for_each(
+        linktimeplugin::RegistrarBase<PluginBase>::begin(),
+        linktimeplugin::RegistrarBase<PluginBase>::end(),
+        [](const auto x) {
+            (*x)().dosomething();
```

Comment:
I find that empty pair of parentheses un-intuitive. I understand why it's needed, but one would expect the iterator to work as usual, i. e. `x->dosomething();`, as in the original example.

Reply:
I agree. I don't like that form at all either. I noticed that down-casting is needed to get the right type object back.